packages/ses/test/test-console-error-trap.js

```diff
@@ -0,0 +1,24 @@
+import test from 'ava';
+import { exec } from 'child_process';
+import { dirname, join } from 'path';
+
+const cwd = join(
+  dirname(new URL(import.meta.url).pathname),
+  'console-error-trap',
+);
+
+test.cb('errors reveal their stacks', t => {
+  t.plan(3);
+  exec('node index.js', { cwd }, (err, stdout, stderr) => {
+    t.assert(err, 'exit code should be non-zero');
```

Comment:
```suggestion
    t.assert(err !== 0, 'exit code should be non-zero');
```
I do sometimes wonder how much we should rely on the reader's knowledge of JS's rules of truthiness and falsiness. I don't have a strong opinion on this one.

Reply:
I like to use the typescript checker as a very careful but not very knowledgeable reader. `tsc` should be told (by the callee) what type is expected. Then it can carefully check that expectations are met. For example:

```js
/** @param {unknown} x */
function truthy(x) { return !!x; }

truthy('abc'); // OK

/** @param {boolean} condition */
function isTrue(condition) { return condition === true; }

isTrue('abc'); // Argument of type 'string' is not assignable to parameter of type 'boolean'.ts(2345)
```

Downsides:
  1. Until we get to a pretty strict "no any" policy (#600), nothing ensures that the callee provided such a declaration and the default is `any`... so callers trained as above might infer from lack of a red squiggle that they're doing it right when in fact lack of a red squiggle just means tsc didn't check it.
  2. Conversely, callees have to keep in mind that type declarations are static advisory checks, not runtime defensive correctness guarantees.

Reply:
I’ll check on the type. I think it’s an `Error` that may have a `code` number property, not a number itself.

Reply:
I’ve addressed this by making the tests more specific.